WordPressKit/DashboardServiceRemote.swift

```diff
@@ -32,11 +32,16 @@ open class DashboardServiceRemote: ServiceRemoteWordPressComREST {
         })
     }
 
-    private func makeQueryParams(cards: [String], deviceId: String) throws -> [String: AnyObject] {
+    private func makeQueryParams(cards: [String], deviceId: String?) throws -> [String: AnyObject] {
         let cardsParams: [String: AnyObject] = [
             "cards": cards.joined(separator: ",") as NSString
         ]
-        let featureFlagParams = try SessionDetails(deviceId: deviceId).dictionaryRepresentation()
+        let featureFlagParams: [String: AnyObject]? = try {
+            guard let deviceId else {
+                return nil
+            }
```

Comment:
Oh, interesting. I didn't realize it would be possible to handle a missing `deviceId` in this way. Looks like a nice way to keep the app "working".